resources/src/trade.rs

```diff
@@ -0,0 +1,253 @@
+    counter_account: String,
+    // counter_asset: AssetIdentifier,
+    price: PriceRatio,
+    seller: Seller,
+}
+
+/// The seller involved in the trade.
+#[derive(Eq, PartialEq, Debug, Copy, Clone)]
+pub enum Seller {
+    /// The base created the sell offer.
+    Base,
+    /// The counter created the sell offer.
+    Counter,
+}
+
+impl Seller {
+    /// Returns whether the seller is the base
+    ///
+    /// ```
+    /// use stellar_resources::TradeSeller;
+    /// let seller = TradeSeller::Base;
+    /// assert!(seller.is_base());
+    /// ```
+    pub fn is_base(&self) -> bool {
+        *self == Seller::Base
+    }
+
+    /// Returns whether the seller is the counter
+    ///
+    /// ```
+    /// use stellar_resources::TradeSeller;
+    /// let seller = TradeSeller::Counter;
+    /// assert!(seller.is_counter());
+    /// ```
+    pub fn is_counter(&self) -> bool {
+        *self == Seller::Counter
+    }
+}
+
+impl<'de> Deserialize<'de> for Trade {
+    fn deserialize<D>(d: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let rep: TradeRepresentation = TradeRepresentation::deserialize(d)?;
+        let seller = if rep.base_is_seller {
+            Seller::Base
+        } else {
+            Seller::Counter
+        };
+        Ok(Trade {
+            id: rep.id,
+            paging_token: rep.paging_token,
+            ledger_close_time: rep.ledger_close_time,
+            offer_id: rep.offer_id,
+            base_account: rep.base_account,
+            base_amount: rep.base_amount,
+            counter_account: rep.counter_account,
+            counter_amount: rep.counter_amount,
+            price: PriceRatio::from(rep.price),
+            seller: seller,
+        })
+    }
+}
+
+impl Serialize for Trade {
+    fn serialize<S>(&self, s: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let rep = TradeRepresentation {
+            id: self.id.to_owned(),
+            paging_token: self.paging_token.to_owned(),
+            ledger_close_time: self.ledger_close_time,
+            offer_id: self.offer_id.to_owned(),
+            base_account: self.base_account.to_owned(),
+            base_amount: self.base_amount,
+            base_asset_type: "".to_string(),
+            base_asset_code: None,
+            base_asset_issuer: None,
+            counter_amount: self.counter_amount,
+            counter_account: self.counter_account.to_owned(),
+            counter_asset_type: "".to_string(),
+            counter_asset_code: None,
+            counter_asset_issuer: None,
+            price: Price {
+                n: self.price.numerator(),
+                d: self.price.denominator(),
+            },
+            base_is_seller: self.seller.is_base(),
+        };
+        rep.serialize(s)
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "UPPERCASE")]
+struct Price {
+    n: u64,
+    d: u64,
+}
+
+impl From<Price> for PriceRatio {
```

Comment:
Should this be in the offer file where PriceRatio is defined?  Not sure of the convention here

Reply:
No, the 'Price' here is private to the trade, so it should only be here.